File: Patrones/Inmutable/Java/Inmutable/src/example/CircleData.java
package example;

public class CircleData {
	private double radius;
	private Color color;
	private Point origin;
	
	public CircleData(double radius, Color color, Point origin) {
		this.radius = radius;
		this.color = color;
	    this.origin = origin;
	}
	
	public double getRadius() {
		return this.radius;
	}
	
	public Color getColor() {
		return this.color;
	}
	
	public Point getOrigin() {
		return this.origin;
	}
}

File: Patrones/Inmutable/C++/src/CircleData.cpp
#include <CircleData.h>

CircleData::CircleData(const double & radius, Color * color, Point * origin) {
	this->radius = radius;
	this->color = color;
	this->origin = origin;	
}

double CircleData::getRadius(){
	return this->radius;
}

Color * CircleData::getColor(){
	return this->color;
}

Point * CircleData::getOrigin(){
	return this->origin;
}

CircleData::~CircleData(){
	delete origin;
	delete color;
}